### PR TITLE
Create data model

### PR DESCRIPTION
**Pull request by Zuhaib042**
### Create data model in Rails Blogs App milestone-01
**Summary**
- In this milestone, I set up the Postgresql database to the rails app.
- Then, I created the schema using Rails migration.
- I focused on the ERD diagram for the project while creating tables.

### CODE REVIEW

File: db/migrate/20230117152518_add_user_ref_to_posts.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToPosts < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :user, null: false, foreign_key: true
```
Comment (TedLivist) [inline db/migrate/20230117152518_add_user_ref_to_posts.rb]: - [ ] The post should have an `author_id`, which means the reference should refer to `:author`, not `:user`. You can point Rails to the user table using the `to_table` option.

  You can look at [this StackOverflow answer](https://stackoverflow.com/a/42056089) on how to run the migration for this.

File: db/migrate/20230117153703_add_user_ref_to_comments.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToComments < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :comments, :user, null: false, foreign_key: true
```
Comment (TedLivist) [inline db/migrate/20230117153703_add_user_ref_to_comments.rb]: - [ ] The comment should have an `author_id`, which means the reference should be to `:author`, not `:user`. You can point Rails to the user table using the `to_table` option.

File: db/migrate/20230117153849_add_user_ref_to_likes.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToLikes < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :likes, :user, null: false, foreign_key: true
```
Comment (TedLivist) [inline db/migrate/20230117153849_add_user_ref_to_likes.rb]: - [ ] The like should have an `author_id`, which means the reference should refer to `:author`, not `:user`. You can point Rails to the user table using the `to_table` option.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Zuhaib042,

Good job so far!
✔️ Good documentation
✔️ Use of postgres for database
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — APPROVED** [review]
Hi Zuhaib, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- linters are passing
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**